src/Compilers/CSharp/Test/Symbol/Symbols/DefaultInterfaceImplementationTests.cs

```diff
@@ -30867,5 +30867,2738 @@ static void Main()
+            Assert.True(compilation1.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+            compilation1.VerifyDiagnostics();
+
+            CompilationReference compilationReference = compilation1.ToMetadataReference();
+            MetadataReference metadataReference = compilation1.EmitToImageReference();
+
+            var compilation2 = CreateStandardCompilation(source2, new[] { compilationReference }, options: TestOptions.DebugExe,
+                                                         parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Latest));
+            Assert.True(compilation2.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+            compilation2.VerifyDiagnostics(expected);
+
+            var compilation3 = CreateStandardCompilation(source2, new[] { metadataReference }, options: TestOptions.DebugExe,
+                                                         parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Latest));
+            Assert.True(compilation3.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+            compilation3.VerifyDiagnostics(expected);
+        }
+
+        [Fact]
+        public void Operators_14()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    public static int operator +(I2 x, I1 y)
+    {
+        System.Console.WriteLine(""I1.+"");
+        return 1;
+    }
+}
+
+public interface I2 : I1
+{
+    public static int operator +(I2 x, I1 y)
+    {
+        System.Console.WriteLine(""I2.+"");
+        return 2;
+    }
+}
+";
+
+            var source2 =
+@"
+class Test2 : I2
+{
+    static void Main()
+    {
+        I1 x = new Test2();
+        I2 y = new Test2();
+        var z = y + x;
+        z = x + y;
+    }
+}
+";
+
+            var expected = new DiagnosticDescription[]
+            {
+                // (8,17): error CS0034: Operator '+' is ambiguous on operands of type 'I2' and 'I1'
+                //         var z = y + x;
+                Diagnostic(ErrorCode.ERR_AmbigBinaryOps, "y + x").WithArguments("+", "I2", "I1").WithLocation(8, 17),
+                // (9,13): error CS0019: Operator '+' cannot be applied to operands of type 'I1' and 'I2'
+                //         z = x + y;
+                Diagnostic(ErrorCode.ERR_BadBinaryOps, "x + y").WithArguments("+", "I1", "I2").WithLocation(9, 13)
+            };
+
+            var compilation1 = CreateStandardCompilation(source1, options: TestOptions.DebugDll,
+                                                         parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.Latest));
+            Assert.True(compilation1.Assembly.RuntimeSupportsDefaultInterfaceImplementation);
+            compilation1.VerifyDiagnostics();
+
+            CompilationReference compilationReference = compilation1.ToMetadataReference();
+            MetadataReference metadataReference = compilation1.EmitToImageReference();
+
+           
```

Comment:
Are we testing shadowing does not depend on the order of base interfaces in the derived interface declaration?
```
interface IA
{
    public static IA operator+(IA x, IB y) => null;
}
interface IB : IA
{
    public static IA operator+(IB x, IA y) => null;
}
interface IC1 : IA, IB { }
interface IC2 : IB, IA { }
```